src/Persistence/MassTransit.DapperIntegration/Configuration/Configuration_JobConsumers.cs

```diff
@@ -0,0 +1,288 @@
+﻿#nullable enable
+namespace MassTransit.Configuration
+{
+    using Microsoft.Extensions.DependencyInjection.Extensions;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Options;
+    using System;
+    using System.Data;
+    using System.Data.Common;
+    using DapperIntegration.JobSagas;
+    using DapperIntegration.Saga;
+    using DapperIntegration.SqlBuilders;
+    using Saga;
+    using Internals;
+    using Microsoft.Data.SqlClient;
+    using Npgsql;
+
+
+    public static class DapperJobSagaRepositoryRegistrationExtensions
+    {
+        /// <summary>
+        /// Adds saga repositories for JobConsumers, using the Dapper repository provider.
+        /// </summary>
+        /// <param name="configurator">The JobConsumer configurator</param>
+        /// <param name="configure">The configuration callback</param>
+        public static IJobSagaRegistrationConfigurator DapperRepository(
+            this IJobSagaRegistrationConfigurator configurator,
+            Action<IDapperJobSagaRepositoryConfigurator> configure = null
+        )
+        {
+            var registrationProvider = new DapperJobSagaRepositoryRegistrationProvider(configure);
+
+            configurator.UseRepositoryRegistrationProvider(registrationProvider);
+
+            return configurator;
+        }
+    }
+
+    public interface IDapperJobSagaRepositoryConfigurator
```

Comment:
One class per file, and do not use whatever underscore naming you're doing here for the file name. file name should match type name.